pkg/cmd/util/clientcmd/factory.go

```diff
@@ -375,8 +375,14 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			return kLogsForObjectFunc(object, options)
 		}
 	}
-	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, showLabels, absoluteTimestamps bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
-		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, showLabels, absoluteTimestamps, columnLabels), nil
+	w.Printer = func(mapping *meta.RESTMapping, options *kubectl.PrintOptions) (kubectl.ResourcePrinter, error) {
+		if mapping != nil {
```

Comment:
After I commented here, and just a few minutes ago, I had a suggestion for the similar `resourceShortFormFor` in the Kubernetes code that would simplify matters here too.
https://github.com/openshift/origin/pull/9334/files/1ff8ec94d186fe3770c06cdece2ed8d5d3d9ef99#r69644282

It's a bit strange, why does this Printer automagically sets its own resource type?
Don't we have code to set the type when printing anyway?